app/services/onlinetesting/phase1/Phase1TestService2.scala

```diff
@@ -122,9 +122,58 @@ trait Phase1TestService2 extends OnlineTestService with Phase1TestConcern2 with
     } yield audit("OnlineTestInvitationProcessComplete", application.userId)
   }
 
+  def resetTest(application: OnlineTestApplication, orderIdToReset: String, actionTriggeredBy: String)
+                (implicit hc: HeaderCarrier, rh: RequestHeader): Future[Unit] = eventSink {
+
+    val (invitationDate, expirationDate) = calcOnlineTestDates(integrationGatewayConfig.phase1Tests.expiryTimeInDays)
+
+    for {
+      // Fetch existing test group that should exist
+      maybeTestGroup <- testRepository2.getTestGroup(application.applicationId)
+      testGroup = maybeTestGroup
+        .getOrElse(throw CannotFindTestGroupByApplicationId(s"appId - ${application.applicationId}"))
+
+      // Extract test that requires reset
+      testToReset = testGroup.tests.find(_.orderId == orderIdToReset)
+        .getOrElse(throw CannotFindTestByOrderId(s"OrderId - $orderIdToReset"))
+      _ = Logger.info(s"testToReset -- $testToReset")
+
+      // Create PsiIds to use for re-invitation
+      psiIds = integrationGatewayConfig.phase1Tests.tests.find {
```

Comment:
Is integrationGatewayConfig a global variable, how are we going to unit test it?